sympy/parsing/tests/test_c_parser.py

```diff
@@ -2544,14 +2541,24 @@ def test_binary_operators():
                     Mod(
                         Mul(
                             Add(
-                                Symbol('a'),
-                                Mul(Integer(-1),
-                                    Symbol('b')
+                                Mod(
+                                    Symbol('a'),
+                                    Symbol('mod')
+                                    ),
+                                Mul(
+                                    Integer(-1),
+                                    Mod(
+                                        Symbol('b'),
+                                        Symbol('mod')
+                                        )
                                     )
                                 ),
-                            Add(
-                                Symbol('a'),
-                                Symbol('b')
+                            Mod(
+                                Add(
+                                    Symbol('a'),
+                                    Symbol('b')
+                                    ),
+                                Symbol('mod')
```

Comment:
I'd also try to revert other changes that are in xfail tests

Reply:
This was an actual error in test. The C parser was producing the correct output, but the output being tested against was wrong.